Api/WMC/WMC1/Request.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace baibaratsky\WebMoney\Api\WMC\WMC1;
+
+use baibaratsky\WebMoney\Api\WMC;
+use baibaratsky\WebMoney\Exception\ApiException;
+use baibaratsky\WebMoney\Request\RequestValidator;
+use baibaratsky\WebMoney\Signer;
+
+/**
+ * Class Request
+ *
+ * @link http://wiki.wmtransfer.com/projects/webmoney/wiki/Interface_WMC1
+ */
+class Request extends WMC\Request
+{
+
+    /** @var string payment/purse */
+    protected $payeePurse;
+
+    /** @var string payment/@currency */
+    protected $currency;
+
+    /** @var float payment/price */
+    protected $price;
+
+    /** @var int payment/phone */
+    protected $phone = '';
+
+    /**
+     * @param string $authType
+     *
+     * @throws ApiException
+     */
+    public function __construct($authType = self::AUTH_CLASSIC)
+    {
+        switch ($authType) {
+            case self::AUTH_CLASSIC:
+                $this->url = 'https://transfer.gdcert.com/ATM/Xml/PrePayment1.ashx';
+                break;
+
+            case self::AUTH_LIGHT:
+                $this->url = 'https://transfer.gdcert.com/ATM/Xml/PrePayment1.ashx';
+                $this->signature = base64_encode(
+                    $this->getSignerWmid() . $this->getCurrency() .
+                    $this->getPayeePurse() . $this->getPhone() .
+                    $this->getPrice()
+                );
+
+                break;
+
+            default:
+                throw new ApiException('This interface doesn\'t support the authentication type given.');
+        }
+
+        parent::__construct($authType);
+    }
+
+    /**
+     * @return array
+     */
+    protected function getValidationRules()
+    {
+        return array(
+            RequestValidator::TYPE_REQUIRED => array(
+                'price', 'currency'
+            )
+        );
+    }
+
+    /**
+     * @return string
+     */
+    public function getData()
+    {
+        $xml = '<w3s.request lang="' . $this->getLang() . '">';
+        $xml .= self::xmlElement('wmid', $this->getSignerWmid());
+        $xml .= '<sign type="' . $this->getAuthTypeNum() . '">' . $this->signature . '</sign>';
+        $xml .= '<payment currency="' . $this->getCurrency() . '">';
+        $xml .= self::xmlElement('purse', $this->getPayeePurse());
+        if (empty($this->getPhone())) $xml .= self::xmlElement('phone', $this->getPhone());
```

Comment:
Because PSR2 and ––––>
```json
    "require": {
        "php": ">=5.3",
        "baibaratsky/php-wmsigner": "2.0.*"
    },
```

`empty(callable())` works only for php>55.4